src/orion/algo/pbt/pbt.py

```diff
@@ -375,7 +392,8 @@ def _fork_lineages(self, num):
                 logger.debug("Skipping trial %s", trial)
                 skipped_trials.append(trial)
                 continue
-
+            # NOTE: This assumes that new_trial is not None.
+            assert new_trial is not None
```

Comment:
I'm adding an assert here because that is exactly what the code was already doing, but written more explicitly.
In this particular case, the next few lines would have crashed at runtime if `self._generate_offspring` had returned `None` for `new_trial`, and if `None` was passed as an argument to `fork(new_trial)`.
 
See what I mean? This PR isn't exactly responsible for fixing existing bugs in PBT or other algorithms. This assert here just makes the existing assumptions more explicit, and easier to identify / debug / solve later.

Reply:
`_generate_offspring` returns `(Trial, Trial)` or `(None, None)` so we should never get to this point with `new_trial = None`.  At least if we keep the assert there should be a message saying `new_trial should not be None if trial_to_branch is not, please report this issue at https://github.com/Epistimio/orion/issues`.

Reply:
Interesting! The annotation that I had for the return value of `_generate_offspring` was `tuple[Trial | None, Trial | None]`, which isn't as tight as it can be!
Instead, I've typed it with `tuple[Trial, Trial] | tuple[None, None]`.

The reason for this assertion is, the type checker isn't yet smart enough to understand that if we went through the if above and that therefore `trial_to_branch` is of type `Trial`, then the second tuple item is also non-None. If it were smart enough to figure that out, then I'd gladly remove the assert. But for now, it isn't.

I think the note above the assert is enough, since the users will never see this. This is just for convenience and the type checker, hence no strings, and no friendly messages. 

"Fixed" in 10835fdfbd9aab74c6f427592a89e7bb721d40e9